tests/components/FilterSearch.test.tsx

```diff
@@ -0,0 +1,256 @@
+    expect(mockedExecuteFilterSearch).toHaveBeenLastCalledWith('J');
+
+    expect(mockedExecuteFilterSearch).toHaveBeenCalledTimes(3);
+  });
+
+  it('does not trigger executeFilterSearch when backspacing on an empty text box', () => {
+    render(<FilterSearch searchFields={[searchFieldsProp]}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+
+    userEvent.type(searchBarElement, '{backspace}');
+    expect(searchBarElement).toHaveValue('');
+
+    expect(mockedExecuteFilterSearch).toHaveBeenCalledTimes(0);
+  });
+
+  it('shows autocomplete suggestions when a character is typed and suggestions are returned by useSynchronizedSearch', () => {
+    render (<FilterSearch searchFields={[searchFieldsProp]}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+    userEvent.type(searchBarElement, 'J');
+    expect(mockedExecuteFilterSearch).toBeCalledWith('J');
+
+    const firstAutocompleteSection= screen.getByText('People');
+    expect(firstAutocompleteSection).toBeDefined();
+
+    const firstAutocompleteSuggestion = screen.getByText('John Doe');
+    expect(firstAutocompleteSuggestion).toBeDefined();
+
+    const secondAutocompleteSuggestion = screen.getByText('Jane Doe');
+    expect(secondAutocompleteSuggestion).toBeDefined();
+
+    const secondAutocompleteSection= screen.getByText('Condiments');
+    expect(secondAutocompleteSection).toBeDefined();
+
+    const thirdAutocompleteSuggestion = screen.getByText('Jam');
+    expect(thirdAutocompleteSuggestion).toBeDefined();
+  });
+
+  it('populates the search bar with selected suggestions when users navigate to populated autocomplete suggestions using arrow keys', () => {
+    render (<FilterSearch searchFields={[searchFieldsProp]}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+    userEvent.type(searchBarElement, 'J');
+
+    const firstAutocompleteSuggestion = screen.getByText('John Doe');
+    expect(firstAutocompleteSuggestion).toBeDefined();
+
+    const secondAutocompleteSuggestion = screen.getByText('Jane Doe');
+    expect(secondAutocompleteSuggestion).toBeDefined();
+
+    userEvent.type(searchBarElement, '{arrowdown}');
+    expect(searchBarElement).toHaveValue('John Doe');
+
+    userEvent.type(searchBarElement, '{arrowdown}');
+    expect(searchBarElement).toHaveValue('Jane Doe');
+  });
+
+  it('calls executeSearch when an autocomplete suggestion is selected using the arrow keys and "Enter"', () => {
+    render (<FilterSearch searchFields={[searchFieldsProp]}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+    userEvent.type(searchBarElement, 'J');
+
+    const firstAutocompleteSuggestion = screen.getByText('John Doe');
+    expect(firstAutocompleteSuggestion).toBeDefined();
+
+    userEvent.type(searchBarElement, '{arrowdown}');
+    expect(searchBarElement).toHaveValue('John Doe');
+    userEvent.type(searchBarElement, '{enter}');
+    expect(setFilterOption).toBeCalled();
```

Comment:
we want to make sure setFilterOption and setOffset were called with the right things, specifically (and respectively) the right Filter and 0

Reply:
it would also be good to check the invocation call order, since we want to make sure setFilterOption and setOffset are both called before executeSearch! You can check this with `setFilterOption.mock.invocationCallOrder` (I only learned you can do this from your pr)

Reply:
added checks to make sure they were called with the correct parameters and called in the correct order.